### PR TITLE
setup ossf scorecard and codql workflows

### PR DESCRIPTION
**Pull request by mmorel-35**
Signed-off-by: Matthieu MOREL <matthieu.morel35@gmail.com>

### CODE REVIEW

**Comment by mmorel-35** [comment]
cc @phlax ,
Noticed that there is a formatter for the yaml files.

File: .github/workflows/scorecard.yml
Comment (phlax) [inline .github/workflows/scorecard.yml:3]: prefer just `contents`

**Review by phlax — COMMENTED** [review]
@mmorel-35 thanks - same style nits - indents/ordering etc - but otherwise lgtm

**Comment by phlax** [comment]
>  Noticed that there is a formatter for the yaml files.

yeah yamllint - we wrap it with `envoy.code.check` - would be good to add that and `glint` to all of the repos (maintainers willing) as we did in toolshed 

https://github.com/envoyproxy/toolshed/blob/fa87bda6b7edd8e426fded6587b5e773e45a026e/.github/workflows/ci.yml#L54-L63

**Comment by mmorel-35** [comment]
@phlax would it be a lack of rights that explains the failure ?

**Comment by phlax** [comment]
seems there are 2 issues - the `build` issue looks a lot to me like a network transient

not sure about `pre-commit` issue but it seems to have gone away in last push

**Comment by phlax** [comment]
i dont think it can be permissions as such - its a `pull_request` trigger which is ~permissionless

**Comment by mmorel-35** [comment]
OK ! Everything seems to work now.
Anything to add to merge this ?

**Comment by phlax** [comment]
could you fix the indents please - othewise lgtm

cc @ysawa0 @mattklein123 would be good to get signoff from a repo maintainer

**Comment by mmorel-35** [comment]
"Fixing" the indent means changing the way it is scope. It seems out of the scope of this PR. 
Can it be done in another PR ?

File: integration-test/Dockerfile.tester
Comment (ysawa0) [inline integration-test/Dockerfile.tester:1]: So moving forward we should always ref a sha for images and such?

**Review by ysawa0 — APPROVED** [review]
This looks great, thank you.

File: integration-test/Dockerfile.tester
Comment (phlax) [reply integration-test/Dockerfile.tester:1]: yeah its generally preferable - but probs we can let dependabot or similar take care of it

**Pull request merged**
